kernel/src/hil/time.rs

```diff
@@ -557,6 +612,21 @@ impl Ticks for Ticks64 {
     fn max_value() -> Self {
         Ticks64(!0u64)
     }
+
+    #[inline]
+    fn from_or_max(val: u64) -> Self {
+        Self(val)
+    }
+
+    #[inline]
+    fn saturating_scale(self, num: u32, den: u32) -> u32 {
+        let scaled = self.0.saturating_mul(num as u64) / den as u64;
+        if scaled < u32::MAX as u64 {
+            scaled as u32
+        } else {
+            u32::MAX
+        }
+    }
```

Comment:
We can only return a u32, so we can't use u64:MAX.

The tick conversion methods before assumed we didn't want to deal with more than u32 for time (which I think is a fine assumption), so I held that assumption in the new conversion methods. We limit all conversions to saturated u32.

Reply:
I understand, the `u32` returned here is only ever in units of time, rather than units of ticks. I think this does make sense then. (Whether time should be specifiable in u64s is a possible discussion, but outside of the scope of this PR.)